src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -450,4 +451,20 @@ AND EXISTS (
                )
         """)
     List<ExternalObjectDirectoryEntity> findExpiredMediaEntries(OffsetDateTime maxRetentionDate, Limit batchSize);
+
+    @Query("""
+        SELECT eod FROM ExternalObjectDirectoryEntity eod
+                WHERE eod.media.id = :mediaId
+                AND eod.status.id = 2
+                AND eod.externalLocationType.id IN (1, 2)
+        """)
+    List<ExternalObjectDirectoryEntity> findByMediaStatusAndLocation(@Param("mediaId") Integer mediaId);
```

Comment:
the name `findByMediaStatusAndLocation` implies that you would be passing in status and location, but you aren't, its hard coded, in which case should the name be something like `findStoredInInboundAndUnstructuredByMediaId

Reply:
done